mu4/notation/internal/notationinteraction.cpp

```diff
@@ -1847,66 +1882,86 @@ void NotationInteraction::swapSelection()
 
 void NotationInteraction::deleteSelection()
 {
-    m_undoStack->prepareChanges();
+    startEdit();
     score()->cmdDeleteSelection();
-    m_undoStack->commitChanges();
+    apply();
 
-    m_selectionChanged.notify();
+    notifyAboutSelectionChanged();
 }
 
 void NotationInteraction::flipSelection()
 {
-    m_undoStack->prepareChanges();
+    startEdit();
     score()->cmdFlip();
-    m_undoStack->commitChanges();
+    apply();
 
-    m_selectionChanged.notify();
+    notifyAboutSelectionChanged();
 }
 
 void NotationInteraction::addTieToSelection()
 {
-    m_undoStack->prepareChanges();
+    startEdit();
     if (noteInput()->isNoteInputMode()) {
         score()->cmdAddTie();
     } else {
         score()->cmdToggleTie();
     }
-    m_undoStack->commitChanges();
+    apply();
 
-    m_selectionChanged.notify();
+    notifyAboutSelectionChanged();
 }
 
 void NotationInteraction::addSlurToSelection()
 {
-    m_undoStack->prepareChanges();
+    startEdit();
     doAddSlur();
-    m_undoStack->commitChanges();
+    apply();
 
-    m_selectionChanged.notify();
+    notifyAboutSelectionChanged();
+}
+
+void NotationInteraction::addOttavaToSelection(OttavaType type)
+{
+    startEdit();
+    score()->cmdAddOttava(type);
+    apply();
+
+    notifyAboutNotationChanged();
```

Comment:
Please explain why are you calling this notify here?

Reply:
removed